dynamiq/storages/vector/dry_run.py

```diff
@@ -0,0 +1,96 @@
+from pydantic import BaseModel, ConfigDict, Field
+
+from dynamiq.utils.logger import logger
+
+
+class DryRunConfig(BaseModel):
+    """Configuration for dry run.
+
+    Attributes:
+        enabled (bool): Whether dry run is enabled. Defaults to False.
+        delete_documents: Whether to delete the ingested documents after the dry run. Defaults to True.
+        delete_collection: Whether to delete the created collection after the dry run. Defaults to False.
+    """
+
+    enabled: bool = False
+    delete_documents: bool = Field(default=True, description="Delete the ingested documents")
+    delete_collection: bool = Field(default=False, description="Delete the created collection")
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+class DryRunMixin:
+    """Mixin class to add dry run functionality to vector stores.
+
+    This mixin provides resource tracking and cleanup capabilities for vector stores
+    operating in dry run mode. It tracks ingested documents and created collections
+    to enable cleanup based on the DryRunConfig settings.
+    """
+
+    def __init__(self, dry_run_config: DryRunConfig | None = None):
+        """Initialize the DryRunMixin.
+
+        Args:
+            dry_run_config: Configuration for dry run behavior. If None, default config is used.
+        """
+        self._dry_run_config = dry_run_config or DryRunConfig()
+        self._tracked_documents: list[str] = []
+        self._tracked_collection: str | None = None
+
+    def delete_documents(self, document_ids: list[str] | None = None, delete_all: bool = False) -> None:
+        """Delete documents by their IDs.
+
+        Args:
+            document_ids: List of document IDs to delete.
+            delete_all: Whether to delete all documents.
+        """
+        raise NotImplementedError("Subclass must implement delete_documents method")
```

Comment:
I think here we need to use a pass, as not all nodes should specify `delete_documents` logic. The same is for `def delete_collection`